test/tstKokkosToolsDistributedAnnotations.cpp

```diff
@@ -37,6 +37,69 @@ BOOST_AUTO_TEST_CASE(is_prefixed_with)
   BOOST_TEST(!isPrefixedWith("Nope::ArborX", "ArborX"));
 }
 
+BOOST_AUTO_TEST_CASE_TEMPLATE(
+    distributed_search_tree_distributed_search_tree_allocations_prefixed,
+    DeviceType, ARBORX_DEVICE_TYPES)
+{
+  Kokkos::Tools::Experimental::set_allocate_data_callback(
+      [](Kokkos::Profiling::SpaceHandle handle, const char *label,
+         void const *ptr, uint64_t size) {
+        std::cout << label << '\n';
+        BOOST_TEST((isPrefixedWith(label, "ArborX::DistributedSearchTree::") ||
+                    isPrefixedWith(label, "ArborX::BVH::") ||
+                    isPrefixedWith(label, "ArborX::Sorting::") ||
+                    isPrefixedWith(label, "Testing::")));
+      });
+
+  { // one leaf per process
+    auto tree = makeDistributedSearchTree<DeviceType>(
+        MPI_COMM_WORLD, {
+                            {{{0, 0, 0}}, {{1, 1, 1}}},
+                        });
+  }
+
+  Kokkos::Tools::Experimental::set_allocate_data_callback(nullptr);
+}
```

Comment:
I'm confused about the purpose of this function. What does it test that the next one does not?

Reply:
We are checking the labels for the allocations in the construction here instead of the labels for the allocations when performing the search.